### PR TITLE
Testing Helpers

### PR DESCRIPTION
**Pull request by brandynbennett**
Mocked parts of NATS to ensure that tests would not be sharing global messaging state. Still using a running nats-server to get schemas and stream information.

Also moved the design to be more instance based rather than class based. This made it easier to keep state isolated and also cache schemas.

### CODE REVIEW

File: spec/polyn_spec.rb [hunk 1]
```diff
@@ -75,33 +74,34 @@
...
         a: 1,
         b: 2,
-      }, store_name: store_name, header: { "a header key" => "a header value" }, reply_to: "foo")
+      }, header: { "a header key" => "a header value" }, reply_to: "foo")
 
       msg = get_message("calc.mult.v1", "my_consumer", "CALC")
 
-      expect(msg.header).to eq({ "Nats-Msg-Id" => JSON.parse(msg.data)["id"], "a header key" => "a header value" })
+      expect(msg.header).to eq({ "Nats-Msg-Id" => JSON.parse(msg.data)["id"],
+"a header key" => "a header value" })
```
Comment (xathien) [inline spec/polyn_spec.rb:96]: Moar nitpick, but this should probably float out a little farther.

**Review by xathien — APPROVED** [review]

File: lib/polyn/configuration.rb [hunk 2]
```diff
@@ -5,10 +5,13 @@ module Polyn
   # Configuration data for Polyn
   class Configuration
     def initialize
-      @domain      = nil
-      @source_root = nil
+      @domain                = nil
+      @source_root           = nil
+      @polyn_env             = ENV["POLYN_ENV"] || ENV["RAILS_ENV"] || "development"
```
[the remark this answers is not in the document]
Comment (brandynbennett) [reply lib/polyn/configuration.rb:10]: My editor did that for me, how helpful. Very weird indeed. I'll try and adjust

File: spec/polyn_spec.rb (hunk 1 above)
Comment (brandynbennett) [reply spec/polyn_spec.rb:96]: :shakes fist at rubocop:

File: lib/polyn/configuration.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (brandynbennett) [reply lib/polyn/configuration.rb]: https://github.com/SpiffInc/polyn-ruby/pull/13/commits/7de0e6791d2df41774fd708fe5507ae1a2235585

File: spec/polyn_spec.rb (hunk 1 above)
Comment (brandynbennett) [reply spec/polyn_spec.rb]: https://github.com/SpiffInc/polyn-ruby/pull/13/commits/9f8f8623931543e0739796a55be9e41c23b46fb1

**Review by xathien — APPROVED** [review]

**Pull request merged**